src/Microsoft.AspNetCore.OData/Adapters/WebApiControllerContext.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Microsoft.AspNet.OData.Adapters
+{
+    /// <summary>
+    /// Adapter class to convert Asp.Net WebApi controller context to OData WebApi.
+    /// </summary>
+    internal class WebApiControllerContext : IWebApiControllerContext
+    {
+        /// <summary>
+        /// The inner context wrapped by this instance.
+        /// </summary>
+        private RouteContext innerContext;
+
+        /// <summary>
+        /// Initializes a new instance of the WebApiControllerContext class.
+        /// </summary>
+        /// <param name="routeContext">The inner context.</param>
+        /// <param name="controllerResult">The selected controller result.</param>
+        public WebApiControllerContext(RouteContext routeContext, SelectControllerResult controllerResult)
+        {
+            if (routeContext == null)
```

Comment:
neednot test the null for controller result?

Reply:
Fixed.